### PR TITLE
Replace strftime with f-strings where nicer

### PR DESCRIPTION
**Pull request by eumiro**

### CODE REVIEW

**Comment by eumiro** [comment]
@uranusjr Reduced to being part of larger f-strings.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
> I’m not particular sure about this, strftime is more obvious and (subjectively) easier to read. The f-string format can be beneficial if the formatted date string is going into another f-string (as in some cases here) but not when it’s on its own.

I really like conciseness of the way it's done now. It very nicely focuses the concerns (date + format) in one place within the larger f-string. @uranusjr -> is that good for you as well?

File: airflow/providers/elasticsearch/log/es_task_handler.py [hunk 1]
```diff
@@ -206,9 +206,7 @@ def _clean_date(value: datetime | None) -> str:
 
         https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#_reserved_characters
         """
-        if value is None:
-            return ""
-        return value.strftime("%Y_%m_%dT%H_%M_%S_%f")
+        return value.strftime("%Y_%m_%dT%H_%M_%S_%f") if value else ""
```
Comment (uranusjr) [inline airflow/providers/elasticsearch/log/es_task_handler.py]: Doesn’t seem particularly necessary to me, we’re not playing code golf…

Comment (uranusjr) [reply airflow/providers/elasticsearch/log/es_task_handler.py]: Same for other early return checks below.

File: tests/providers/apache/kylin/operators/test_kylin_cube.py [hunk 2]
```diff
@@ -37,8 +37,8 @@ class TestKylinCubeOperator:
         "project": "learn_kylin",
         "cube": "kylin_sales_cube",
         "command": "build",
-        "start_time": datetime(2012, 1, 2, 0, 0).strftime("%s") + "000",
-        "end_time": datetime(2012, 1, 3, 0, 0).strftime("%s") + "000",
+        "start_time": f"{datetime(2012, 1, 2, 0, 0):%s}000",
+        "end_time": f"{datetime(2012, 1, 3, 0, 0):%s}000",
```
Comment (uranusjr) [inline tests/providers/apache/kylin/operators/test_kylin_cube.py]: It’s probably better to just convert these to literals with a comment, going through datetime just to convert into a datetime is a bit over the top.

Comment (eumiro) [reply tests/providers/apache/kylin/operators/test_kylin_cube.py]: Changed it to a more explicit version.

File: airflow/providers/elasticsearch/log/es_task_handler.py (hunk 1 above)
Comment (eumiro) [reply airflow/providers/elasticsearch/log/es_task_handler.py]: Reverted all such cases.

**Review by potiuk — APPROVED** [review]

**Pull request merged**